### DISCUSSION TITLE
Same model does not load again

### DISCUSSION THREAD

**Discussion by iamvaibhavrai** [brianzinn/react-babylonjs#266] [Q&A] [answered]
I am trying to load the model in a scene by passing the URL and File Name. It loads appropriately for the first time but when I try to load it again, it won't.

```
<Model
    name={sceneName.split('.')[0]}
    reportProgress
    position={new Vector3(0, 0, 0)}
    rootUrl={sceneURL}
    sceneFilename={sceneName}
    scaleToDimension={400.0}
    onModelLoaded={(model) => onLoaded(model)}
    id="venue"
/>

const [sceneURL, setSceneURL] = useState('');
const [sceneName, setSceneName] = useState('');

setSceneName(sceneName);
setSceneURL(sceneURL);
```

I have made a state for sceneURL and sceneName and setting their values on click of different buttons.

- Click on button A to Load model A
- Click on button B to Load model B
- Again click on button A to load model A (it doesn't work). It doesn't make a call to the network to download the model. Also, onLoaded isn't called.

**Comment by brianzinn** [comment, marked as answer]
does this work:
```jsx
<Model key={`${sceneName}-${sceneURL}`} ... />
```

There are a few different recipes that have been shared and depends on your scenario how best to get it working.  I think I will add a `Model` component that doesn't use Suspense, since it suits some use-cases better.

**Reply by jinincarnate** [reply]
```
  const [counter, setCounter] = useState(0);

useEffect(() => {
.
.
setCounter(counter + 1);
.
}
  });

<Model key={counter}..../>
```

used something like this and it worked

**Reply by jinincarnate** [reply]
but when it replaces the previous model, it deletes it associated materials and textures, so to preserve them i explicitly disposed the model without disposing its materials and textures.
